### ISSUE TITLE
dials.image_viewer isoresolution lines not available for 2theta!=0

### ISSUE THREAD

**Issue by Anthchirp** [dials/dials#66]
dials.image_viewer resolution on mouseover does not match the drawn resolution rings on images taken with 2theta!=0.

I noticed that the resolution rings are not distorted to ellipses as one would expect, so probably the mouseover value is more trustworthy.

example file:

```
$ dials.image_viewer /dls/i19-1/data/2016/mt11397-3/TKR948/TKR948_06_01302.cbf
```

Mouseover resolutions on the ring marked 1.14A go from 1.4 (right) and 1.5 (top/bottom) to 2.7 (left)

**Comment by rjgildea** [comment]
The resolution rings are meaningless at 2theta != 0, so should be ignored.
They assume that the detector is perpendicular to the beam.

**Comment by Anthchirp** [comment]
So this would be a thing for the future then to provide resolution indicators. I23 might be interested as well.

**Comment by graeme-winter** [comment]
Following discussion just now hyperbolic resolution lines will be a thing for 45<|2theta|<135 degrees; give we know the principle axes (and are free to select definitions of x, y) we should be able to plot these correctly.

**Comment by Anthchirp** [comment]
Ellipses now appear to work properly. Next step hyperbolas, then multi-panel detectors.

**Comment by Anthchirp** [comment]
The demand for hyperbolas seems rather limited, so the work required to display those is not proportionate at this time.

**Comment by biochem-fan** [comment]
I reopen this old issue because 2theta != 0 configurations are quite common with RIGAKU's in house diffractometers (Synergy with HyPix) and small molecular crystallography. It is also necessary to support non flat detectors. For example, panels in the CITIUS detector being developed at SACLA/SPring-8 are at different depths; outer rows are shallower than inner rows.

TODO:
- support different camera lengths for each panel (draw arcs instead of a circle/ellipse)
- support hyperbola

**Comment by biochem-fan** [comment]
> different camera lengths for each panel

This has been already available in `calculate_isoresolution_lines` in `util/image_viewer/spotfinder_frame.py`. Because of the way panels are projected in the image viewer, I missed it. Sorry for my false alarm.

> support hyperbola

This should be also possible with `calculate_isoresolution_lines` but somehow nothing is displayed in my RIGAKU Synergy datasets. I will check.

**Comment by biochem-fan** [comment]
It turned out to be a simple logic error. I will submit a pull request later today.

**Comment by biochem-fan** [comment]
I found this line is problematic:

https://github.com/dials/dials/blob/9c17e694fe6b31459478a361b1a55b49fe5c623f/src/dials/util/image_viewer/spotfinder_frame.py#L895-L903

The comment sounds like `if detector.has_projection_2d` was true for non-coplanar detectors but
it is true only when `set_projection_2d()` is explicitly called in the dxtbx format class. In addition, `_draw_resolution_polygons` should also be used for planar detectors when 2theta >> 0, because the normal code path supports only circles (2theta = 0) and ellipses (small 2theta) but not hyperbola.

I would like to suggest we discard the old code path and always use `_draw_resolution_polygons`.

By doing so (i.e. setting `if True`), resolution lines on HyPix images are displayed properly.
Without this, no lines are shown at all.

```
wget https://data.pdbjlc1.pdbj.org/xrda/94/frames/exp_864_22_1.rodhypix
dials.image_viewer exp_864_22_1.rodhypix show_resolution_rings=True
```

<img width="217" alt="Image" src="https://github.com/user-attachments/assets/5436011e-3bd5-4386-bf52-78df27789ece" />

This also fixes resolution rings for our CITIUS detector, whose rows are at different depths.
Note the gap between panel 45 and 46. The latter is about 2.3 mm closer to the crystal.

<img width="616" alt="Image" src="https://github.com/user-attachments/assets/b11f8bcb-d769-477f-9fc3-d3d2420a2807" />

@dagewa you are the author of `draw_resolution_polygons`. Do you agree this change?
The only drawback is that lines are less smooth for circles and ellipses.
But writing a proper test for large 2theta and/or non-flat detectors is tricky.

**Comment by dagewa** [comment]
> I would like to suggest we discard the old code path and always use _draw_resolution_polygons.

Yes, I agree. I think it's another instance where we have tried to be be clever in the past, but ended up with two different ways of doing things and a more complicated codebase. The dumb polygon approximation is pretty good for most visual comparison purposes. The smoothness could be improved by providing access to `n_rays` here

https://github.com/dials/dials/blob/9c17e694fe6b31459478a361b1a55b49fe5c623f/src/dials/util/image_viewer/spotfinder_frame.py#L118-L126

At the moment this is hardcoded, but I suppose we could make this a user-adjustable parameter if the (lack of) smoothness bothers anyone.
